doc/api/fs.md

```diff
@@ -5625,6 +5637,50 @@ The following constants are meant for use with the [`fs.Stats`][] object's
   </tr>
 </table>
 
+### Dirent Constants
+
+The following constants are meant for use with the [`fs.Dirent`][] class'
+`type` argument for determining the type of entity being represented.
+
+<table>
+  <tr>
+    <th>Constant</th>
+    <th>Description</th>
+  </tr>
+  <tr>
+    <td><code>UV_DIRENT_UNKNOWN</code></td>
+    <td>An unknown entity type.</td>
+  </tr>
+  <tr>
+    <td><code>UV_DIRENT_FILE</code></td>
+    <td>Entity is a regular file.</td>
+  </tr>
+  <tr>
+    <td><code>UV_DIRENT_DIR</code></td>
+    <td>Entity is a directory.</td>
+  </tr>
+  <tr>
+    <td><code>UV_DIRENT_LINK</code></td>
+    <td>Entity is a symbolic link.</td>
+  </tr>
+  <tr>
+    <td><code>UV_DIRENT_FIFO</code></td>
+    <td>Entity is a FIFO or pipe.</td>
+  </tr>
+  <tr>
+    <td><code>UV_DIRENT_SOCKET</code></td>
+    <td>Entity is a socket.</td>
+  </tr>
+  <tr>
+    <td><code>UV_DIRENT_BLOCK</code></td>
+    <td>Entity is a block-oriented device file (a device like a disk).</td>
+  </tr>
+  <tr>
+    <td><code>UV_DIRENT_CHAR</code></td>
```

Comment:
Documenting these makes no sense because they're not exposed, they live in `internalBinding('constants')`.

Reply:
How do you mean that they are not exposed? They are accessible through `require('fs').constants.UV_DIRENT_CHAR` like the other constants.

I.e. on Win10 x64 Node.JS 12.13.1 the constants object contains the following:

```js
{
  UV_FS_SYMLINK_DIR: 1,
  UV_FS_SYMLINK_JUNCTION: 2,
  O_RDONLY: 0,
  O_WRONLY: 1,
  O_RDWR: 2,
  UV_DIRENT_UNKNOWN: 0,
  UV_DIRENT_FILE: 1,
  UV_DIRENT_DIR: 2,
  UV_DIRENT_LINK: 3,
  UV_DIRENT_FIFO: 4,
  UV_DIRENT_SOCKET: 5,
  UV_DIRENT_CHAR: 6,
  UV_DIRENT_BLOCK: 7,
  S_IFMT: 61440,
  S_IFREG: 32768,
  S_IFDIR: 16384,
  S_IFCHR: 8192,
  S_IFLNK: 40960,
  O_CREAT: 256,
  O_EXCL: 1024,
  UV_FS_O_FILEMAP: 536870912,
  O_TRUNC: 512,
  O_APPEND: 8,
  F_OK: 0,
  R_OK: 4,
  W_OK: 2,
  X_OK: 1,
  UV_FS_COPYFILE_EXCL: 1,
  COPYFILE_EXCL: 1,
  UV_FS_COPYFILE_FICLONE: 2,
  COPYFILE_FICLONE: 2,
  UV_FS_COPYFILE_FICLONE_FORCE: 4,
  COPYFILE_FICLONE_FORCE: 4
}
```

Reply:
Yeah, that seems like a regrettable oversight... I traced it back to dcccbfdc799 - that commit should have been more discriminating in what it re-exports. Everything prefixed with `UV_` has no business being there.